### PR TITLE
Change scheme to specify supervisor spec

### PR DESCRIPTION
**Pull request by ttokit**
Since Supervisor.Spec is deprecated, changed to use streamlined child spec

### CODE REVIEW

**Comment by skirino** [comment]
Thank you for the PR!
Could you remove `1.4.5` in `.travis.yml` as we no longer support 1.4.x?

**Comment by ttokit** [comment]
OK, removed in 83b3464

File: lib/pool_sup.ex [hunk 1]
```diff
@@ -40,6 +40,18 @@ defmodule PoolSup do
   #
   # client API
   #
+  @doc """
+  Returns a specification to start this module under a supervisor.
+  """
+  defun child_spec(args :: list) :: map do
```
Comment (skirino) [inline lib/pool_sup.ex]: Can we use `Supervisor.child_spec` as the return type here?

Comment (ttokit) [reply lib/pool_sup.ex]: Yes, it looks more specific and clarity!
fixed in a4f2c45

File: lib/pool_sup.ex [hunk 2]
```diff
@@ -40,6 +40,18 @@ defmodule PoolSup do
   #
   # client API
   #
+  @doc """
+  Returns a specification to start this module under a supervisor.
```
Comment (skirino) [inline lib/pool_sup.ex]: Let's change this doc comment (and the corresponding one for `Multi`) to something like:
```
Returns a child specification to be used when it's not fully specified by the parent supervisor.
```

Comment (ttokit) [reply lib/pool_sup.ex]: fixed in d429f41

**Pull request merged**